torch/distributed/fsdp/_flat_param.py

```diff
@@ -2229,6 +2229,9 @@ def _writeback_orig_params(self) -> bool:
                     f"Expects to have saved tensor for {flat_param._fqns[i]}",
                 )
             param_changed = getattr(module, param_name) is not param
+            from torch.distributed._tensor import DTensor
```

Comment:
This import probably should be at the top of the function.

Reply:
I recall there was some internal torch package issues where this particular file should not default import `DTensor` or else something breaks. That is why `DTensor` is imported inline here too:
https://github.com/pytorch/pytorch/blob/ae6d206f05c32b8dbcfb34c660db9089e98bba38/torch/distributed/fsdp/_flat_param.py#L1886

Reply:
I moved to the top of the function, but not default import